src/api/controllers/cache-controller.ts

```diff
@@ -1,18 +1,29 @@
 import { RequestHandler, Request, Response } from 'express';
 import * as prom from 'prom-client';
-import { FoundOrNot, logger } from '../../helpers';
-import { DataStore, DbChainTip } from '../../datastore/common';
+import { logger } from '../../helpers';
+import { DataStore } from '../../datastore/common';
 import { asyncHandler } from '../async-handler';
 
 const CACHE_OK = Symbol('cache_ok');
-const CHAIN_TIP_LOCAL = 'chain_tip';
 
 // A `Cache-Control` header used for re-validation based caching.
 // `public` == allow proxies/CDNs to cache as opposed to only local browsers.
 // `no-cache` == clients can cache a resource but should revalidate each time before using it.
 // `must-revalidate` == somewhat redundant directive to assert that cache must be revalidated, required by some CDNs
 const CACHE_CONTROL_MUST_REVALIDATE = 'public, no-cache, must-revalidate';
 
+/**
+ * Describes a key-value to be saved into a request's locals, representing the current
+ * state of the chain depending on the type of information being requested by the endpoint.
+ * This entry will have an `ETag` string as the value.
+ */
+export enum ETagType {
+  /** ETag based on the latest `index_block_hash` or `microblock_hash`. */
+  chainTip = 'chain_tip',
+  /** ETag based on a digest of all pending mempool `tx_id`s. */
+  mempool = 'mempool',
+}
+
 interface ChainTipCacheMetrics {
   chainTipCacheHits: prom.Counter<string>;
```

Comment:
May be useful to expand this to include counter specific to an etag/cache type (mempool vs chaintip)